libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -624,6 +624,12 @@ void GCS_MAVLINK::ftp_list_dir(struct pending_ftp &request, struct pending_ftp &
 
     request.data[sizeof(request.data) - 1] = 0; // ensure the path is null terminated
 
+    // Strip trailing /
```

Comment:
this change is not needed with the fixes in my branch

Reply:
I think we still do, its a actually a issue with the this function. If for example I do a get of `scripts/` the `gen_dir_entry` then adds another `/` here: https://github.com/ArduPilot/ardupilot/blob/e01e697343cf3eb9b0991d7153f135b60b96f965/libraries/GCS_MAVLink/GCS_FTP.cpp#L603

So it ends up with a full path of `scripts//test.lua` the Posix file system copes fine with that, but ROMFS falls over. I have not tested others.

Reply:
ok, makes sense